pkg/cloud/mesos/mesos.go

```diff
@@ -72,9 +89,30 @@ func (c *MesosCloud) Zones() (cloudprovider.Zones, bool) {
 
 // Mesos does not provide support for multiple clusters
 func (c *MesosCloud) Clusters() (cloudprovider.Clusters, bool) {
-	//TODO(jdef): we could probably implement this and always return a
-	//single cluster- this one.
-	return nil, false
+	return c, true
+}
+
+// List lists the names of the available clusters.
+func (c *MesosCloud) ListClusters() ([]string, error) {
+	// Always returns a single cluster (this one!)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	name, err := c.client.clusterName(ctx)
+	return []string{name}, err
+}
+
+// Master gets back the address (either DNS name or IP address) of the master node for the cluster.
+func (c *MesosCloud) Master(clusterName string) (string, error) {
+	clusters, err := c.ListClusters()
+	if err != nil {
+		return "", err
+	}
+	for _, name := range clusters {
+		if name == clusterName {
+			return c.client.master, nil
```

Comment:
`c.client.master` may be empty if there's no current leader

also, see : http://golang.org/pkg/net/#SplitHostPort